### PR TITLE
[BACKLOG-43001] [BISERVER-15167] [BISERVER-15168] Restoring schedules from previous and current version

### PR DESCRIPTION
**Pull request by singletonc**
Worked with @dcleao and @nawaz34-hitachivantara to address the fact that schedules were failing to import due to user roles/permissions, and output path validation for different importing user vs schedule user and vfs:
- Made sure to be getting the user from the schedule request, not the session
- Use the schedule request user for output path validation via runAs
- Refactored and updated UT coverage

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 3m 35s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]
Only issue is formatting.

**Comment by buildguy** [comment]
### :x: Build failed in 2m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 8m 2s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 10 test(s) failed:

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenFallbackDoesNotExistThenFallsBackFarther</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenOutputFolderHasNoAccessThenFallsBack</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_ContainsPatternAlready</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenNoAvailableFolderOrFallbackReturnsNull</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenFallbackHasNoAccessThenFallsBackFarther</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_withJobName</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenOutputFolderDoesNotExistThenFallsBack</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenGenericFileServiceThrowsCheckingOutputFolderThenFallsBack</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

<details><summary><b>org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolverTest.testResolveOutputFilePath_whenNoOutputFileThenFallsBack</b> (click to expand)</summary><p>

```
!EnterpriseSchedulerService.InvalidUserName!
```

</p></details>

  
Tests run: **234**, Failures: **10**, Skipped: **10** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-242/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 8m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **234**, Failures: **0**, Skipped: **10** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-242/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **234**, Failures: **0**, Skipped: **10** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-242/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerOutputPathResolver.java [hunk 1]
```diff
@@ -96,11 +102,43 @@ public String getJobName() {
...
+  private String getScheduleOwnerFromRequest() {
+    String userName = scheduleRequest.getJobParameters().stream()
+      .filter( jobParameter -> StringUtils.equals( ( jobParameter ).getName(), IScheduler.RESERVEDMAPKEY_ACTIONUSER ) )
+      .findFirst()
+      .map( jobParameter -> jobParameter.getValue() != null ? ( (String) jobParameter.getValue() ).trim() : null )
+      .orElse( null );
+
+    // Schedule Owner should have already been validated with a fallback to administrator
+    // if there is a problem with the schedule owner at this point, it is a bigger issue.
+    if ( userName == null || userName.isEmpty() ) {
```
Comment (NJtwentyone) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerOutputPathResolver.java:123]: Are there any test scenarios catching/asserting `IllegalArgumentException` in SchedulerOutputPathResolverTest.java:

- userName == null
- username.isEmpty()

Comment (singletonc) [reply core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerOutputPathResolver.java:123]: updated UTs  with some refactoring of repeated code and added coverage, more notes in commit message
https://github.com/pentaho/pentaho-scheduler-plugin/pull/242/commits/dc1e0c05aa778e20d7241093392be3e4e3ee4148

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 53s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **236**, Failures: **0**, Skipped: **10** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-242/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **236**, Failures: **0**, Skipped: **10** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-242/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Review by dcleao — APPROVED** [review]

**Review by NJtwentyone — APPROVED** [review]

**Review by rmansoor — APPROVED** [review]

**Pull request merged**